src/lib/y2partitioner/widgets/encrypt.rb

```diff
@@ -0,0 +1,172 @@
+      # options each time the encryption method is changed.
+      #
+      # @macro seeCustomWidget
+      def handle(event)
+        if event["ID"] == encrypt_method_widget.widget_id
+          encrypt_method_options_widget.refresh(encrypt_method_widget.value)
+        end
+
+        nil
+      end
+
+      # @macro seeCustomWidget
+      def help
+        format(
+          _("<p>%{header}</p><ul>%{content}</ul>"),
+          header:  help_header,
+          content: help_content
+        )
+      end
+
+      private
+
+      # @return [Actions::Controllers::Encryption] controller for the encryption device
+      attr_reader :controller
+
+      # Widgets to show
+      #
+      # @return [Array<Widgets>]
+      def widgets
+        widgets = []
+        widgets << encrypt_method_widget if display_encrypt_method?
+        widgets << encrypt_method_options_widget
+
+        widgets
+      end
+
+      # Returns the widget to select the encryption method
+      #
+      # @return [Widgets::EncryptMethod]
+      def encrypt_method_widget
+        @encrypt_method_widget ||= EncryptMethod.new(controller)
+      end
+
+      # Returns the widget in charge of displaying the options for the selected
+      # encryption method
+      #
+      # @return [Widgets::EncryptMethodOptions]
+      def encrypt_method_options_widget
+        @encrypt_method_options_widget ||= EncryptMethodOptions.new(controller)
+      end
+
+      # Whether the encryption method widget should be displayed
+      #
+      # @return [Boolean] true if there is more than one method available; false otherwise
+      def display_encrypt_method?
+        controller.several_encrypt_methods?
+      end
+
+      # The introductory help text
+      #
+      # @return [String]
+      def help_header
+        if controller.several_encrypt_methods?
+          _("The following encryption methods can be chosen:")
+        else
+          _("The following encryption method will be used:")
+        end
+      end
+
+      # Help texts for available encryption methods
+      #
+      # @return [String]
+      def help_content
+        texts = controller.methods.map do |m|
+          text = send("help_for_#{m.id}", m)
+          "<li>#{text}</li>"
+        end
+
+        texts.join
+      end
+
+      # Help text for the Random Swap encryption method
+      #
+      # @return [String]
+      def help_for_random_swap(encrypt_method)
+        format(
+          _("<p><b>%{label}</b>: this encryption method uses randomly generated keys at boot and it " \
+            "will not support Hibernation to hard disk. The swap device is re-encrypted during every " \
+            "boot, and its previous content is destroyed. You should disable Hibernation through your " \
+            "respective DE Power Management Utility and set it to Shutdown on Critical to avoid Data " \
+            "Loss!</p>" \
+            "<p>Also note the device UUID changes every time the swap is encrypted, and the file " \
```

Comment:
I would reorganize the information in this paragraph a bit. Something like:

Please, make sure your swap device is mounted by a stable name that is not subject to change on every reboot (e.g. for a swap partition use the udev device id instead of the partition device name). Otherwise a wrong device could be encrypted instead of your swap! In that regard, note both the file system label and the UUID change every time the swap is re-encrypted, so they are not valid options to mount a randomly encrypted swap device.

Reply:
The only think I disagree is "the file system label and the UUID change every time ". Label does not change every time, but it is not set. Well, we can see it as it changes the first time to "my label" to "". But the rest of times it is always "". What about saying something like the label is not set?

Reply:
I see the difference, but I wanted to keep the explanation simple. Saying "both change so they are not reliable" sounded more straightforward than "one changes and the other is bumped so none of them are reliable for their own reasons".

In other words, I found reasonable to sacrifice precision in order to keep the message straight. But I don't have a strong opinion, TBH.